opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -140,3 +140,147 @@ const fn severity_of_level(level: &Level) -> Severity {
+        // play well with unit tests.
+        let _guard = tracing::subscriber::set_default(subscriber);
+
+        // Act
+        error!(name: "my-event-name", target: "my-system", event_id = 20, user_name = "otel", user_email = "otel@opentelemetry.io");
+        logger_provider.force_flush();
+
+        // Assert TODO: move to helper methods
+        let exported_logs = exporter
+            .get_emitted_logs()
+            .expect("Logs are expected to be exported.");
+        assert_eq!(exported_logs.len(), 1);
+        let log = exported_logs
+            .get(0)
+            .expect("Atleast one log is expected to be present.");
+
+        // Validate common fields
+        assert_eq!(log.instrumentation.name, "opentelemetry-appender-tracing");
+        assert_eq!(log.record.severity_number, Some(Severity::Error));
+
+        // Validate trace context is none.
+        assert!(log.record.trace_context.is_none());
+
+        // Validate attributes
+        let attributes: Vec<(Key, AnyValue)> = log
+            .record
+            .attributes
+            .clone()
+            .expect("Attributes are expected");
+        assert_eq!(attributes.len(), 4);
+        assert!(attributes.contains(&(Key::new("name"), "my-event-name".into())));
+        assert!(attributes.contains(&(Key::new("event_id"), 20.into())));
+        assert!(attributes.contains(&(Key::new("user_name"), "otel".into())));
+        assert!(attributes.contains(&(Key::new("user_email"), "otel@opentelemetry.io".into())));
+    }
+
+    #[test]
+    fn tracing_appender_inside_tracing_context() {
+        // Arrange
+        let exporter: InMemoryLogsExporter = InMemoryLogsExporter::default();
+        let logger_provider = LoggerProvider::builder()
+            .with_simple_exporter(exporter.clone())
+            .build();
+
+        let layer = layer::OpenTelemetryTracingBridge::new(&logger_provider);
+        let subscriber = tracing_subscriber::registry().with(layer);
+
+        // avoiding setting tracing subscriber as global as that does not
+        // play well with unit tests.
+        let _guard = tracing::subscriber::set_default(subscriber);
+
+        // setup tracing as well.
+        let tracer_provider = TracerProvider::builder()
+            .with_config(config().with_sampler(Sampler::AlwaysOn))
+            .build();
+        let tracer = tracer_provider.tracer("test-tracer");
+
+        let mut trace_id_expected = TraceId::from_hex("1").unwrap();
+        let mut span_id_expected = SpanId::from_hex("1").unwrap();
+
+        // Act
+        tracer.in_span("test-span", |cx| {            
+            trace_id_expected = cx.span().span_context().trace_id();
+            span_id_expected = cx.span().span_context().span_id();
+
+            // logging is done inside span context.
+            error!(name: "my-event-name", target: "my-system", event_id = 20, user_name = "otel", user_email = "otel@opentelemetry.io");
+        });
```

Comment:
nit - As [tracer::in_span](https://github.com/open-telemetry/opentelemetry-rust/blob/cd23ec279ad8954001e7b855f3132c2030312453/opentelemetry/src/trace/tracer.rs#L204) method returns the value produced by closure, we can use that to initialize  `trace_id_expected` and `span_id_expected`, something like below:

```rust
let (trace_id_expected, span_id_expected) = tracer.in_span("test-span", |cx| {
     trace_id = cx.span().span_context().trace_id();
     span_id = cx.span().span_context().span_id();
     // error() ...
    (trace_id, span_id)
});
```